test_netcdf_timeaverage.jl

```diff
@@ -0,0 +1,213 @@
+                                timestepper = :RungeKutta3,
+                                tracers = (:c1, :c2),
+                                forcing = (c1=c1_forcing, c2=c2_forcing))
+
+    set!(model, c1=1, c2=1)
+    simulation = Simulation(model, Δt=Δt, stop_time=stop_time)
+
+    ∫c1_dxdy = Field(Average(model.tracers.c1, dims=(1, 2)))
+    ∫c2_dxdy = Field(Average(model.tracers.c2, dims=(1, 2)))
+        
+    nc_outputs = Dict("c1" => ∫c1_dxdy, "c2" => ∫c2_dxdy)
+    nc_dimensions = Dict("c1" => ("zC",), "c2" => ("zC",))
+
+    single_time_average_nc_filepath = "single_decay_windowed_time_average_test.nc"
+    
+    window = window_nΔt*Δt
+    interval = interval_nΔt*Δt
+
+    single_nc_output = Dict("c1" => ∫c1_dxdy)
+    single_nc_dimension = Dict("c1" => ("zC",))
+
+    simulation.output_writers[:single_output_time_average] =
+        NetCDFOutputWriter(model, single_nc_output,
+                           array_type = Array{Float64},
+                           verbose = false,
+                           filename = single_time_average_nc_filepath,
+                           schedule = AveragedTimeInterval(interval, window = window, stride = stride),
+                           dimensions = single_nc_dimension,
+                           overwrite_existing = overwrite)
+    checkpointer = Checkpointer(model,
+                                schedule = TimeInterval(stop_time),
+                                prefix = "test",
+                                cleanup = true)
+
+    simulation.output_writers[:checkpointer] = checkpointer
+
+
+    function progress_message(sim) 
+        wta = sim.output_writers[:single_output_time_average].outputs["c1"]
+        wd = wta.window_start_time
+        ws = wta.window_start_iteration
+        pc = wta.previous_collection_time
+        fo = wta.fetch_operand
+        res = sum(wta.result)
+        mct = model.clock.time
+        interval = wta.schedule.interval
+        window = wta.schedule.window
+        cll = wta.schedule.collecting
+        nat = next_actuation_time(wta.schedule)
+        fat = wta.schedule.first_actuation_time
+        actuation =  wta.schedule.actuations
+        @info string("Iter: ", iteration(sim), ", time: ", prettytime(sim), ", model clock time:", mct, ", collecting: ", cll, ", u-avg: ", res, ", window_start_time: ", wd, ", window_start_iteration: ", ws, ", previous_collection_time:", pc, ", fetch_operand: ", fo, ", next_actuation_time: ", nat, ", first_actuation_time: ", fat, ", actuation: ", actuation, ", interval:", interval, ", window:", window)
+    end
+        simulation.callbacks[:progress] = Callback(progress_message, TimeInterval(Δt))
+
+
+    return simulation
+
+end
+    
+
+
+function next_actuation_time(sch::AveragedTimeInterval)
+    t₀ = sch.first_actuation_time
+    N = sch.actuations
+    interval = sch.interval
+    return t₀ + (N + 1) * interval 
+    # the actuation time is the end of the time averaging window
+end
```

Comment:
Why is this being defined in the test script?

Reply:
Maybe there are better ways of doing this, but in order to print out useful information related to the actuation to the progress message, I find it convenient to define this function in this script.

Reply:
Isn't this function in the source code?

Reply:
yeah, exactly.

Reply:
So, instead of redefining it here, you can simply use the one in the source code